### DISCUSSION TITLE
Plausible Analytics - Setting up one application on two subdomains

### DISCUSSION THREAD

**Discussion by Luxapodular** [plausible/analytics#482] [Self-Hosted Support]
Hi Plausible Community!

I run www.wickeditor.com. I currently have a React Application that I host on 2 separate subdomains that I'd like to hook up to Plausible.

[editor.wickeditor.com](editor.wickeditor.com) - Main Application
[test.wickeditor.com](test.wickeditor.com)- Test Application

My issue is that the Main and Test Application are both built from the same source. What would be the best way to structure my plausible script so that it properly runs on both my test. and editor. subdomains?

Is it proper to simply add both scripts to the index.html? Or, should I be doing this in another fashion?

**Comment by Vigasaurus** [comment]
Are they both compiling from the same React app? How are you choosing the split off the dist/ folder? Both running the same build, with react-router or similar routing based on subdomain? Or with generating different dist/'s based on a compilation tag? (I probably could be checked the link but heh, I'm lazy). 

If they are both the same source exactly, I'd probably recommend something on your host-side to inject the right script based on which deploy you're on - what's your host?

Also, are you wanting to split both out to different plausible dashboards? Or both to the same?

**Reply by Luxapodular** [reply]
Honestly, you writing out these questions gave me a bit more clarity on how I may fix this problem.

So, I've got a single repository that builds to the same /build folder and then copies that build to an 'editor' or 'test' repository separately. The reason for this, is to allow github pages to serve both projects (GH Pages only allows a single repo to have a custom domain, so I can't have a separate repo be a sites subdirectory). In this case, injecting the correct script to the index.html's head, based on the build script I'm using seems like the right play.

I think the better solution is actually to fix my project up a bit by serving 'editor' and 'test' as sub directories, rather than as subdomains... A much trickier solution than I had initially hoped.

I'd preferably like the subdomains on the same plausible dashboard so that I can truly see the number of unique visitors, including to subdomains. e.g. www.wickeditor.com, editor.wickeditor.com, and test.wickeditor.com. But I know this isn't possible currently without some cookie-based solution.

Thank you for the reply by the way, made this much clearer!

**Reply by ukutaht** [reply]
A simple thing you can try is instead of including the script like this:

```html
<script async defer data-domain="hardcoded-domain.com" src="https://plausible.io/js/plausible.js"></script>
```
Try something like:

⚠️⚠️ Untested pseudo-code ahead ⚠️⚠️⚠️

```html
<script>
var script = document.createElement('script')
script.async=true
script.defer=true
script.dataset.domain=window.location.host // Set script data-domain dynamically
script.src = "https://plausible.io/js/plausible.js"
document.head.append(script)
</script>
```

**Reply by robinpecha** [reply]
> ```
> <script>
> var script = document.createElement('script')
> script.async=true
> script.defer=true
> script.dataset.domain=window.location.host // Set script data-domain dynamically
> script.src = "https://plausible.io/js/plausible.js"
> document.head.append(script)
> </script>
> ```

Wooooov, it works like a charm. Thanks for saving me.
